### ISSUE TITLE
Windows Hosting Bundle should upgrade in place within a major.minor product band

### ISSUE THREAD

**Issue by dagood** [aspnet/AspNetCore#7729]
When I install a new patch version of the hosting bundle, old patch versions with the same major and minor versions should be uninstalled. For example, installing `3.0.1` should uninstall `3.0.0`.

This is part of implementing in-place upgrade across Windows .NET Core installers: https://github.com/dotnet/core-setup/issues/4188.

https://github.com/dotnet/core-sdk/pull/530 implements this for the .NET Core SDK bundle, and the same work is planned for the .NET Core Runtime bundle.

@natemcmaster @joeloff

**Comment by joeloff** [comment]
I chatted with @Eilon and @muratg, will probably do this after preview 3 since the deadline is today and doesn't provide enough time to fully test the changes

**Comment by joeloff** [comment]
PR is almost complete. I'm just waiting on the final preview 3 build to drop

**Comment by Eilon** [comment]
@joeloff - is the Hosting Bundle work for this already done in https://github.com/aspnet/AspNetCore/pull/8450 ? Should this bug be closed?

**Comment by joeloff** [comment]
I have a few more tests to run, then I'll close it. Probably Monday if that's okay

**Comment by Eilon** [comment]
Sounds good. Monday @ noon we branch, but if it's just tests, just keep checking into master, and you're good to go.

**Comment by joeloff** [comment]
Testing looks good. I've done a couple of upgrades from the preview1 and 2 hosting bundles to the preview4 build.
